assets/js/components/ExecutionResults/ExecutionResultsPage.jsx

```diff
@@ -29,13 +29,13 @@ function ExecutionResultsPage() {
   } = useSelector(getLastExecutionData(clusterID));
 
   useEffect(() => {
-    if (catalog.length === 0) {
-      dispatch(updateCatalog());
+    if (cluster?.provider) {
```

Comment:
Would you declare this like `const cloudProvider = cluster?.provider;` and just reference this instead of accessing two times the field?